src/MSBuildBinLogQuery/Component/Build.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Concurrent;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.Build.Logging.Query.Component
+{
+    public class Build : Component
+    {
+        public ConcurrentDictionary<int, Project> Projects { get; }
+        public ConcurrentBag<Project> OrderedProjects { get; }
+
+        public Build() : base()
+        {
+            Projects = new ConcurrentDictionary<int, Project>();
+            OrderedProjects = new ConcurrentBag<Project>();
+        }
+
+        public Project GetOrAddProject(int id, ProjectStartedEventArgs args)
```

Comment:
If you drop OrderedProjects this can be simplified 

```C#
Projects.GetOrAdd(id, (_) => new Project(
                id,
                args.ProjectFile,
                args.TargetNames,
                args.Items,
                args.Properties,
                args.GlobalProperties,
                this));
```

Reply:
Done.